### PR TITLE
Extend post category features.

### PR DESCRIPTION
**Pull request by RajibSarker867602**
-- Completed Tasks:
1.  Create Post -> Should show all the existing category----**(done)**
2.  Add additional property in Category table to set property as "Show In Homepage Menu"----**(done)**
3.  Add button/checkbox to set this value in admin panel----**(done)**
4. Only Admin can update this----**(done)**

### CODE REVIEW

File: src/SpotLights/Api/CategoryController.cs [hunk 1]
```diff
@@ -15,53 +15,67 @@ namespace SpotLights.Interfaces;
...
 
-    [HttpDelete("{idsString}")]
-    public async Task DeleteAsync([FromRoute] string idsString)
-    {
-        IEnumerable<int> ids = idsString.Split(',').Select(int.Parse);
-        await _categoryService.DeleteAsync<Category>(ids);
-    }
+  [HttpDelete("{id:int}")]
+  public async Task DeleteAsync([FromRoute] int id)
+  {
```
Comment (anupsarkar-dev) [inline src/SpotLights/Api/CategoryController.cs]: In backend you should check if the request is made by Admin user or not

File: src/SpotLights/Api/CategoryController.cs [hunk 2]
```diff
@@ -15,53 +15,67 @@ namespace SpotLights.Interfaces;
...
+  }
 
-    [HttpGet("{postId:int}")]
-    public async Task<IEnumerable<Category>> GetPostCategories(int postId)
-    {
-        return await _categoryService.GetPostCategories(postId);
-    }
+  [HttpDelete("{idsString}")]
+  public async Task DeleteAsync([FromRoute] string idsString)
+  {
```
Comment (anupsarkar-dev) [inline src/SpotLights/Api/CategoryController.cs]: In backend you should check if the request is made by Admin user or not

File: src/SpotLights/Api/CategoryController.cs [hunk 3]
```diff
@@ -15,53 +15,67 @@ namespace SpotLights.Interfaces;
...
+    await _categoryService.DeleteAsync<Category>(id);
+  }
 
-    [HttpGet("{postId:int}")]
-    public async Task<IEnumerable<Category>> GetPostCategories(int postId)
-    {
-        return await _categoryService.GetPostCategories(postId);
-    }
+  [HttpDelete("{idsString}")]
+  public async Task DeleteAsync([FromRoute] string idsString)
```
Comment (anupsarkar-dev) [inline src/SpotLights/Api/CategoryController.cs]: rename idsString to ids only

File: src/SpotLights/Api/CategoryController.cs [hunk 4]
```diff
@@ -15,53 +15,67 @@ namespace SpotLights.Interfaces;
...
+
+  [HttpPut]
+  public async Task<ActionResult<bool>> SaveCategory(Category category)
+  {
+    return await _categoryService.SaveCategory(category);
+  }
+
+  [HttpPut("updateCategoryMenuStatus/{categoryId}")]
+  public async Task<ActionResult<bool>> UpdateCategoryMenuStatus(int categoryId, [FromBody] bool status)
+  {
```
Comment (anupsarkar-dev) [inline src/SpotLights/Api/CategoryController.cs]: should check if the request is made by Admin user or not

File: src/SpotLights/appsettings.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "SpotLights": {
     "DbProvider": "Mssql", // Sqlite, Postgres
-    "ConnString": "Data Source=.\\MSSQL2022;Initial Catalog=SpotLights2;Integrated Security=True;Encrypt=False;Trust Server Certificate=True",
+    "ConnString": "Data Source=.\\MSSQL2022;Initial Catalog=SpotLights;Integrated Security=True;Encrypt=False;Trust Server Certificate=True",
```
Comment (anupsarkar-dev) [inline src/SpotLights/appsettings.json:4]: do not push your local connection string changes

**Review by anupsarkar-dev — CHANGES_REQUESTED** [review]

**Comment by RajibSarker867602** [comment]
-- UI changes:

- Show/Hide button added in Categories list page
image

- Dropdown list added for categories in post create page
image

**Pull request merged**